### PR TITLE
feat: build darwin amd64/arm64 Docker images

### PR DESCRIPTION
**Pull request by kuisathaverat**
build darwin amd64/arm64 Docker images

depends on https://github.com/elastic/golang-crossbuild/pull/197
related to https://github.com/elastic/golang-crossbuild/issues/153

### CODE REVIEW

File: go/darwin-arm64/Dockerfile.tmpl [hunk 1]
```diff
@@ -34,7 +34,7 @@ RUN tar -xzf /tmp/osxcross.tar.gz -C / \
 
 ENV PATH $PATH:$OSXCROSS_PATH/bin
 # Add osxcross libraries to the library PATH
-ENV LD_LIBRARY_PATH ${OSXCROSS_PATH}/lib:$LD_LIBRARY_PATH
+ENV LD_LIBRARY_PATH $OSXCROSS_PATH/lib:/lib:/usr/lib:/usr/local/lib:$LD_LIBRARY_PATH
```
Comment (kuisathaverat) [inline go/darwin-arm64/Dockerfile.tmpl:37]: force the whole LD_LIBRARY_PATH

File: go/darwin/Dockerfile.tmpl [hunk 2]
```diff
@@ -51,23 +51,35 @@ RUN \
 
 ENV PATH $PATH:$OSXCROSS_PATH/bin
 # Add osxcross libraries to the library PATH
-ENV LD_LIBRARY_PATH ${OSXCROSS_PATH}/lib:$LD_LIBRARY_PATH
+ENV LD_LIBRARY_PATH $OSXCROSS_PATH/lib:/lib:/usr/lib:/usr/local/lib:$LD_LIBRARY_PATH
```
Comment (kuisathaverat) [inline go/darwin/Dockerfile.tmpl:54]: force the whole LD_LIBRARY_PATH

File: go/darwin/Dockerfile.tmpl [hunk 3]
```diff
@@ -51,23 +51,35 @@ RUN \
...
 RUN cd / \
-  && o32-clang helloWorld.c -o helloWorld \
-  && file helloWorld \
-  && file helloWorld | grep -c 'Mach-O i386' \
-  && rm helloWorld.c helloWorld
+  && o32-clang helloWorld.c -o helloWorld.i368 \
+  && file helloWorld.i368 \
+  && file helloWorld.i368 | grep -c 'Mach-O i386' \
+  && rm helloWorld.*
 {{- end }}
```
Comment (kuisathaverat) [inline go/darwin/Dockerfile.tmpl:83]: refactor tests

**Review by v1v — APPROVED** [review]

File: go/darwin-arm64/Makefile [hunk 4]
```diff
@@ -1 +1,8 @@
 include ../Makefile.common
+
+ifeq ($(DOCKER_MULTIARCH),1)
+DOCKER_CMD := $(SELF_DIR)/../.ci/scripts/buildx.sh
+
+push:
+	@echo "Already done."
```
Comment (kuisathaverat) [inline go/darwin-arm64/Makefile:7]:
```suggestion
	@echo "Already done by buildx (.ci/scripts/buildx.sh)."
```

File: go/darwin/Makefile [hunk 5]
```diff
@@ -1 +1,8 @@
 include ../Makefile.common
+
+ifeq ($(DOCKER_MULTIARCH),1)
+DOCKER_CMD := $(SELF_DIR)/../.ci/scripts/buildx.sh
+
+push:
+	@echo "Already done."
```
Comment (kuisathaverat) [inline go/darwin/Makefile:7]:
```suggestion
```suggestion
	@echo "Already done by buildx (.ci/scripts/buildx.sh)."
atomic-push:
	@echo "Already done by buildx (.ci/scripts/buildx.sh)."
```
```

**Pull request merged**